File: 0009. Palindrome Number/PalindromeNumber.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        reverse = 0
        if x < 0:
        	return False
        elif x == 0:
        	return True
        else:
        	xx = x
        	while xx > 0:
        		n = xx % 10
        		reverse = reverse * 10 + n
        		xx = xx // 10
        	if reverse == x:
        		return True
        	else:
        		return False

File: 0009. Palindrome Number/PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
    	int reverse = 0;
        if (x < 0) {
        	return false;
        } else if (x == 0) {
        	return true;
        } else {
            int xx = x;
        	while (xx > 0) {
        		int n = xx % 10;
        		reverse = reverse * 10 + n;
        		xx /= 10;
        	}
        	if (reverse == x) {
        		return true;
        	} else {
        		return false;
        	}
        }
    }
}
